src/HDC1080.c

```diff
@@ -48,7 +50,14 @@ float GetTemperature(void)
 	uint8_t regaddr = HDC1080_TEMP_REG;
 	uint8_t regval[2];
 
-	I2C_IF_ReadFrom(HDC1080_DEV_ADDR, &regaddr, 1, regval, 2);
+	I2C_IF_Write(HDC1080_DEV_ADDR, &regaddr, 1, 0);
+
```

Comment:
As per the data sheet, I think temp needs some conversion:

 temp = (temp / 0x10000) * 165 - 40